src/ctr_exit.c

```diff
@@ -151,6 +152,7 @@ void do_exit_command()
 		_pexit("Failed to reset signal for SIGCHLD");
 	}
 
+	remove_g_unix_fds();
```

Comment:
This does seem like a potential bug, but I'm not sure it's the actual bug here.  This problem can only occur if we reenter `g_main_loop_run()` after closing a fd behind its back.

I am not seeing where we'd being doing that.  It looks like inside `do_exit_command()` here we do some prep stuff and then just `execve()`.

Are you sure there's not another place where we're closing a fd while we're inside the `g_main_loop_run()` invocation?